<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Siobhon Egan" />

<meta name="date" content="2021-04-02" />

<title>Phyloseq</title>

<script src="site_libs/header-attrs-2.7.2/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">wildlife-bacteria</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="QIIME2.html">QIIME2</a>
</li>
<li>
  <a href="phyloseq.html">Phyloseq</a>
</li>
<li>
  <a href="microbiome-viz.html">Microbiome viz</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/siobhon-egan/wildlife-bacteria">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Phyloseq</h1>
<h4 class="author">Siobhon Egan</h4>
<h4 class="date">04/02/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>wildlife-bacteria/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210129code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210129code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210129)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsiobhoneganwildlifebacteriatreeebe20d01dc71c1ba2eb30c7ab053c4d2d0c78821targetblankebe20d0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/siobhon-egan/wildlife-bacteria/tree/ebe20d01dc71c1ba2eb30c7ab053c4d2d0c78821" target="_blank">ebe20d0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsiobhoneganwildlifebacteriatreeebe20d01dc71c1ba2eb30c7ab053c4d2d0c78821targetblankebe20d0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/siobhon-egan/wildlife-bacteria/tree/ebe20d01dc71c1ba2eb30c7ab053c4d2d0c78821" target="_blank">ebe20d0</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  NCBI_data/
    Untracked:  analysis/microbiome-viz-extra.Rmd
    Untracked:  analysis/microbiome-viz.Rmd
    Untracked:  analysis/phylogeny.Rmd
    Untracked:  analysis/site-map.Rmd
    Untracked:  analysis/tois.Rmd
    Untracked:  code/taxa_summary.R
    Untracked:  data/dada2/
    Untracked:  data/dada2_tois/
    Untracked:  data/ps_bact_samp.RData
    Untracked:  data/ps_dec_bact.RData
    Untracked:  data/ps_raw_bact.RData
    Untracked:  data/rickettsia.fasta
    Untracked:  data/site_map.xlsx
    Untracked:  data/taxa_of_interest.RData
    Untracked:  data/taxa_trees/
    Untracked:  data/tmp/
    Untracked:  output/plots/
    Untracked:  tmp/

Unstaged changes:
    Modified:   README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/phyloseq.Rmd</code>) and HTML (<code>docs/phyloseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/siobhon-egan/wildlife-bacteria/blob/ebe20d01dc71c1ba2eb30c7ab053c4d2d0c78821/analysis/phyloseq.Rmd" target="_blank">ebe20d0</a>
</td>
<td>
siobhon-egan
</td>
<td>
2021-02-26
</td>
<td>
add qiime and phyloseq pages
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Visualization and analysis of microbiome data.</p>
<blockquote>
<p>Note this was run using R version 4.0.3 and RStudo version 1.4. See <a href="#rinfo">R info</a> for full details of R session.</p>
</blockquote>
<div id="load-libraries" class="section level2">
<h2>0. Load libraries</h2>
<p><strong>Install libraries if required</strong></p>
<p>Only need to run this code once.</p>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
# phyloseq
source(&#39;http://bioconductor.org/biocLite.R&#39;)
biocLite(&#39;phyloseq&#39;)
#tidyverse
install. packages(&quot;tidyverse&quot;)
#ampvis2
install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;MadsAlbertsen/ampvis2&quot;)
#ampvis2extras
install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;kasperskytte/ampvis2extras&quot;)
#ggpubr
install.packages(&quot;ggpubr&quot;)
#agricolae
install.packages(&quot;agricolae&quot;)
install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;DanielSprockett/reltools&quot;)
devtools::install_github(&#39;jsilve24/philr&#39;)
#decontam
BiocManager::install(&quot;decontam&quot;)
library(decontam)
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;Biostrings&quot;)</code></pre>
<p><strong>Load libraries</strong></p>
<pre class="r"><code>library(qiime2R)
library(phyloseq)
library(tidyverse)
library(ampvis2)
library(ampvis2extras)
library(ggpubr)
library(agricolae)
library(plotly)
library(viridis)
library(cowplot)
library(MicrobeR)
library(microbiome)
library(reshape)
library(decontam)
library(data.table)
library(ape)
library(DESeq2)
library(vegan)
library(microbiomeutilities)
library(knitr)
library(tibble)
library(dplyr)
library(patchwork)
library(Biostrings)</code></pre>
</div>
<div id="import-data" class="section level2">
<h2>1. Import data</h2>
<p>Generate phyloseq object from spreadsheets.</p>
<p>Import ASV/OTU count data</p>
<pre class="r"><code>dada2_ASVs &lt;- read_tsv(&quot;../data/dada2/phyloseq/feature-table.tsv&quot;, skip = 1)
dada2_ASVs_lab = column_to_rownames(dada2_ASVs, var = &quot;#OTU ID&quot;)
# Make matrix
otumat &lt;- as.matrix(dada2_ASVs_lab)</code></pre>
<p>Import taxonomy data</p>
<pre class="r"><code>taxonomy &lt;- read_csv(&quot;../data/dada2/phyloseq/taxonomy.csv&quot;)
taxonomy_lab = column_to_rownames(taxonomy, var = &quot;#q2:types&quot;)
taxmat &lt;- as.matrix(taxonomy_lab)</code></pre>
<p>Check the class of the otumat and taxmat objects, they MUST be in matrix format. Then we can great a phyloseq object called physeq from the otu and taxonomy tables and check the sample names.</p>
<pre class="r"><code>class(otumat)
class(taxmat)
OTU = otu_table(otumat, taxa_are_rows = TRUE)
TAX = tax_table(taxmat)
physeq = phyloseq(OTU, TAX)
physeq
sample_names(physeq)</code></pre>
<p><strong>Add metadata and sequence data</strong></p>
<p>Add sequences to phyloseq object</p>
<pre class="r"><code># read sequence file
rep.seqs &lt;- Biostrings::readDNAStringSet(&quot;../data/dada2/phyloseq/all_rep-seqs.fasta&quot;, format = &quot;fasta&quot;)</code></pre>
<p>Add metadata, importing gDNAID as factor to be able to merge later on</p>
<pre class="r"><code>library(readr)
metadata &lt;- read_csv(&quot;data/dada2/phyloseq/metadata.csv&quot;, 
    col_types = cols(gDNAID = col_factor(levels = c(&quot;categorical&quot;, 
        &quot;Bl_1&quot;, &quot;Bl_10&quot;, &quot;Bl_102&quot;, &quot;Bl_113&quot;, 
        &quot;Bl_114&quot;, &quot;Bl_115&quot;, &quot;Bl_116&quot;, &quot;Bl_117&quot;, 
        &quot;Bl_118&quot;, &quot;Bl_119&quot;, &quot;Bl_120&quot;, &quot;Bl_121&quot;, 
        &quot;Bl_122&quot;, &quot;Bl_123&quot;, &quot;Bl_124&quot;, &quot;Bl_125&quot;, 
        &quot;Bl_126&quot;, &quot;Bl_127&quot;, &quot;Bl_128&quot;, &quot;Bl_129&quot;, 
        &quot;Bl_130&quot;, &quot;Bl_131&quot;, &quot;Bl_132&quot;, &quot;Bl_133&quot;, 
        &quot;Bl_134&quot;, &quot;Bl_135&quot;, &quot;Bl_136&quot;, &quot;Bl_137&quot;, 
        &quot;Bl_138&quot;, &quot;Bl_139&quot;, &quot;Bl_140&quot;, &quot;Bl_141&quot;, 
        &quot;Bl_142&quot;, &quot;Bl_143&quot;, &quot;Bl_144&quot;, &quot;Bl_145&quot;, 
        &quot;Bl_146&quot;, &quot;Bl_147&quot;, &quot;Bl_149&quot;, &quot;Bl_150&quot;, 
        &quot;Bl_151&quot;, &quot;Bl_152&quot;, &quot;Bl_153&quot;, &quot;Bl_154&quot;, 
        &quot;Bl_155&quot;, &quot;Bl_156&quot;, &quot;Bl_157&quot;, &quot;Bl_158&quot;, 
        &quot;Bl_159&quot;, &quot;Bl_160&quot;, &quot;Bl_161&quot;, &quot;Bl_162&quot;, 
        &quot;Bl_163&quot;, &quot;Bl_164&quot;, &quot;Bl_18&quot;, &quot;Bl_19&quot;, 
        &quot;Bl_25&quot;, &quot;Bl_27&quot;, &quot;Bl_28&quot;, &quot;Bl_31&quot;, 
        &quot;Bl_37&quot;, &quot;Bl_43&quot;, &quot;Bl_49&quot;, &quot;Bl_52&quot;, 
        &quot;Bl_54&quot;, &quot;Bl_55&quot;, &quot;Bl_6&quot;, &quot;Bl_61&quot;, 
        &quot;Bl_63&quot;, &quot;Bl_66&quot;, &quot;Bl_7&quot;, &quot;Bl_73&quot;, 
        &quot;Bl_75&quot;, &quot;Bl_78&quot;, &quot;Bl_79&quot;, &quot;Bl_80&quot;, 
        &quot;Bl_82&quot;, &quot;Bl_86&quot;, &quot;Bl_94&quot;, &quot;Bl_98&quot;, 
        &quot;Bl_NTC1&quot;, &quot;Bl_NTC2&quot;, &quot;Bl_NTC3&quot;, 
        &quot;Bl_NTC4&quot;, &quot;Bl_NTC45&quot;, &quot;Bl_NTC46&quot;, 
        &quot;Bl_NTC47&quot;, &quot;Bl_100&quot;, &quot;Bl_101&quot;, &quot;Bl_103&quot;, 
        &quot;Bl_104&quot;, &quot;Bl_105&quot;, &quot;Bl_106&quot;, &quot;Bl_108&quot;, 
        &quot;Bl_109&quot;, &quot;Bl_11&quot;, &quot;Bl_110&quot;, &quot;Bl_111&quot;, 
        &quot;Bl_112&quot;, &quot;Bl_12&quot;, &quot;Bl_13&quot;, &quot;Bl_14&quot;, 
        &quot;Bl_15&quot;, &quot;Bl_16&quot;, &quot;Bl_17&quot;, &quot;Bl_2&quot;, 
        &quot;Bl_20&quot;, &quot;Bl_21&quot;, &quot;Bl_22&quot;, &quot;Bl_23&quot;, 
        &quot;Bl_24&quot;, &quot;Bl_26&quot;, &quot;Bl_29&quot;, &quot;Bl_3&quot;, 
        &quot;Bl_30&quot;, &quot;Bl_32&quot;, &quot;Bl_33&quot;, &quot;Bl_34&quot;, 
        &quot;Bl_35&quot;, &quot;Bl_36&quot;, &quot;Bl_38&quot;, &quot;Bl_39&quot;, 
        &quot;Bl_4&quot;, &quot;Bl_40&quot;, &quot;Bl_41&quot;, &quot;Bl_42&quot;, 
        &quot;Bl_44&quot;, &quot;Bl_45&quot;, &quot;Bl_46&quot;, &quot;Bl_47&quot;, 
        &quot;Bl_48&quot;, &quot;Bl_5&quot;, &quot;Bl_50&quot;, &quot;Bl_51&quot;, 
        &quot;Bl_53&quot;, &quot;Bl_56&quot;, &quot;Bl_57&quot;, &quot;Bl_58&quot;, 
        &quot;Bl_59&quot;, &quot;Bl_60&quot;, &quot;Bl_62&quot;, &quot;Bl_64&quot;, 
        &quot;Bl_65&quot;, &quot;Bl_67&quot;, &quot;Bl_68&quot;, &quot;Bl_69&quot;, 
        &quot;Bl_70&quot;, &quot;Bl_71&quot;, &quot;Bl_72&quot;, &quot;Bl_74&quot;, 
        &quot;Bl_76&quot;, &quot;Bl_77&quot;, &quot;Bl_8&quot;, &quot;Bl_81&quot;, 
        &quot;Bl_83&quot;, &quot;Bl_84&quot;, &quot;Bl_85&quot;, &quot;Bl_87&quot;, 
        &quot;Bl_88&quot;, &quot;Bl_89&quot;, &quot;Bl_9&quot;, &quot;Bl_90&quot;, 
        &quot;Bl_91&quot;, &quot;Bl_92&quot;, &quot;Bl_93&quot;, &quot;Bl_95&quot;, 
        &quot;Bl_96&quot;, &quot;Bl_97&quot;, &quot;Bl_99&quot;, &quot;Bl_C001Q&quot;, 
        &quot;Bl_C002M&quot;, &quot;Bl_C002Q&quot;, &quot;Bl_C003M&quot;, 
        &quot;Bl_C003Q&quot;, &quot;Bl_exbM&quot;, &quot;Bl_exbQ&quot;, 
        &quot;Bl_Q001M&quot;, &quot;Bl_Q001Q&quot;, &quot;Tis_indexNTC1&quot;, 
        &quot;TIS_indexNTC2&quot;, &quot;Tis_indexNTC3&quot;, 
        &quot;Tis_NTC5&quot;, &quot;Tis_NTC54&quot;, &quot;Tis_NTC55&quot;, 
        &quot;Tis_NTC57&quot;, &quot;Tis_NTC58&quot;, &quot;Tis_NTC59&quot;, 
        &quot;Tis_NTC6&quot;, &quot;Tis_NTC60&quot;, &quot;Tis_NTC7&quot;, 
        &quot;Tis_NTC8&quot;, &quot;Tis_104&quot;, &quot;Tis_105&quot;, 
        &quot;Tis_106&quot;, &quot;Tis_107&quot;, &quot;Tis_108&quot;, 
        &quot;Tis_109&quot;, &quot;Tis_110&quot;, &quot;Tis_111&quot;, 
        &quot;Tis_112&quot;, &quot;Tis_113&quot;, &quot;Tis_114&quot;, 
        &quot;Tis_115&quot;, &quot;Tis_116&quot;, &quot;Tis_117&quot;, 
        &quot;Tis_118&quot;, &quot;Tis_119&quot;, &quot;Tis_120&quot;, 
        &quot;Tis_121&quot;, &quot;Tis_122&quot;, &quot;Tis_123&quot;, 
        &quot;Tis_124&quot;, &quot;Tis_125&quot;, &quot;Tis_126&quot;, 
        &quot;Tis_127&quot;, &quot;Tis_128&quot;, &quot;Tis_129&quot;, 
        &quot;Tis_130&quot;, &quot;Tis_131&quot;, &quot;Tis_132&quot;, 
        &quot;Tis_133&quot;, &quot;Tis_134&quot;, &quot;Tis_135&quot;, 
        &quot;Tis_136&quot;, &quot;Tis_137&quot;, &quot;Tis_138&quot;, 
        &quot;Tis_139&quot;, &quot;Tis_140&quot;, &quot;Tis_141&quot;, 
        &quot;Tis_142&quot;, &quot;Tis_143&quot;, &quot;Tis_144&quot;, 
        &quot;Tis_145&quot;, &quot;Tis_146&quot;, &quot;Tis_147&quot;, 
        &quot;Tis_148&quot;, &quot;Tis_149&quot;, &quot;Tis_15&quot;, &quot;Tis_150&quot;, 
        &quot;Tis_151&quot;, &quot;Tis_152&quot;, &quot;Tis_153&quot;, 
        &quot;Tis_154&quot;, &quot;Tis_155&quot;, &quot;Tis_156&quot;, 
        &quot;Tis_157&quot;, &quot;Tis_158&quot;, &quot;Tis_159&quot;, 
        &quot;Tis_160&quot;, &quot;Tis_161&quot;, &quot;Tis_162&quot;, 
        &quot;Tis_163&quot;, &quot;Tis_164&quot;, &quot;Tis_165&quot;, 
        &quot;Tis_166&quot;, &quot;Tis_167&quot;, &quot;Tis_168&quot;, 
        &quot;Tis_169&quot;, &quot;Tis_170&quot;, &quot;Tis_171&quot;, 
        &quot;Tis_172&quot;, &quot;Tis_173&quot;, &quot;Tis_174&quot;, 
        &quot;Tis_175&quot;, &quot;Tis_176&quot;, &quot;Tis_177&quot;, 
        &quot;Tis_178&quot;, &quot;Tis_179&quot;, &quot;Tis_21&quot;, &quot;Tis_43&quot;, 
        &quot;Tis_5&quot;, &quot;Tis_55&quot;, &quot;Tis_56&quot;, &quot;Tis_58&quot;, 
        &quot;Tis_62&quot;, &quot;Tis_68&quot;, &quot;Tis_70&quot;, &quot;Tis_72&quot;, 
        &quot;Tis_80&quot;, &quot;Tis_84&quot;, &quot;Tis_94&quot;, &quot;Tis_95&quot;, 
        &quot;Tis_96&quot;, &quot;Tis_97&quot;, &quot;Tis_SC009&quot;, 
        &quot;Tis_1&quot;, &quot;Tis_10&quot;, &quot;Tis_100&quot;, &quot;Tis_101&quot;, 
        &quot;Tis_102&quot;, &quot;Tis_103&quot;, &quot;Tis_11&quot;, &quot;Tis_12&quot;, 
        &quot;Tis_13&quot;, &quot;Tis_14&quot;, &quot;Tis_16&quot;, &quot;Tis_17&quot;, 
        &quot;Tis_18&quot;, &quot;Tis_19&quot;, &quot;Tis_2&quot;, &quot;Tis_20&quot;, 
        &quot;Tis_22&quot;, &quot;Tis_23&quot;, &quot;Tis_24&quot;, &quot;Tis_25&quot;, 
        &quot;Tis_26&quot;, &quot;Tis_27&quot;, &quot;Tis_28&quot;, &quot;Tis_29&quot;, 
        &quot;Tis_3&quot;, &quot;Tis_30&quot;, &quot;Tis_31&quot;, &quot;Tis_32&quot;, 
        &quot;Tis_33&quot;, &quot;Tis_34&quot;, &quot;Tis_35&quot;, &quot;Tis_36&quot;, 
        &quot;Tis_37&quot;, &quot;Tis_38&quot;, &quot;Tis_39&quot;, &quot;Tis_4&quot;, 
        &quot;Tis_40&quot;, &quot;Tis_41&quot;, &quot;Tis_42&quot;, &quot;Tis_44&quot;, 
        &quot;Tis_45&quot;, &quot;Tis_46&quot;, &quot;Tis_47&quot;, &quot;Tis_48&quot;, 
        &quot;Tis_49&quot;, &quot;Tis_50&quot;, &quot;Tis_51&quot;, &quot;Tis_52&quot;, 
        &quot;Tis_53&quot;, &quot;Tis_54&quot;, &quot;Tis_57&quot;, &quot;Tis_59&quot;, 
        &quot;Tis_6&quot;, &quot;Tis_60&quot;, &quot;Tis_61&quot;, &quot;Tis_63&quot;, 
        &quot;Tis_64&quot;, &quot;Tis_65&quot;, &quot;Tis_66&quot;, &quot;Tis_67&quot;, 
        &quot;Tis_69&quot;, &quot;Tis_7&quot;, &quot;Tis_71&quot;, &quot;Tis_73&quot;, 
        &quot;Tis_74&quot;, &quot;Tis_75&quot;, &quot;Tis_76&quot;, &quot;Tis_77&quot;, 
        &quot;Tis_78&quot;, &quot;Tis_79&quot;, &quot;Tis_8&quot;, &quot;Tis_81&quot;, 
        &quot;Tis_82&quot;, &quot;Tis_83&quot;, &quot;Tis_85&quot;, &quot;Tis_86&quot;, 
        &quot;Tis_87&quot;, &quot;Tis_88&quot;, &quot;Tis_89&quot;, &quot;Tis_9&quot;, 
        &quot;Tis_90&quot;, &quot;Tis_91&quot;, &quot;Tis_92&quot;, &quot;Tis_93&quot;, 
        &quot;Tis_98&quot;, &quot;Tis_99&quot;, &quot;Tis_SC010&quot;, 
        &quot;Tis_SC011&quot;, &quot;Tis_SCexb&quot;, &quot;Tk_1&quot;, 
        &quot;Tk_100&quot;, &quot;Tk_101&quot;, &quot;Tk_102&quot;, &quot;Tk_103&quot;, 
        &quot;Tk_107&quot;, &quot;Tk_108&quot;, &quot;Tk_119&quot;, &quot;Tk_120&quot;, 
        &quot;Tk_122&quot;, &quot;Tk_131&quot;, &quot;Tk_134&quot;, &quot;Tk_136&quot;, 
        &quot;Tk_141&quot;, &quot;Tk_143&quot;, &quot;Tk_144&quot;, &quot;Tk_148&quot;, 
        &quot;Tk_149&quot;, &quot;Tk_150&quot;, &quot;Tk_151&quot;, &quot;Tk_152&quot;, 
        &quot;Tk_153&quot;, &quot;Tk_154&quot;, &quot;Tk_155&quot;, &quot;Tk_156&quot;, 
        &quot;Tk_157&quot;, &quot;Tk_158&quot;, &quot;Tk_159&quot;, &quot;Tk_160&quot;, 
        &quot;Tk_161&quot;, &quot;Tk_162&quot;, &quot;Tk_163&quot;, &quot;Tk_164&quot;, 
        &quot;Tk_165&quot;, &quot;Tk_166&quot;, &quot;Tk_167&quot;, &quot;Tk_168&quot;, 
        &quot;Tk_169&quot;, &quot;Tk_17&quot;, &quot;Tk_170&quot;, &quot;Tk_171&quot;, 
        &quot;Tk_172&quot;, &quot;Tk_173&quot;, &quot;Tk_174&quot;, &quot;Tk_175&quot;, 
        &quot;Tk_176&quot;, &quot;Tk_177&quot;, &quot;Tk_178&quot;, &quot;Tk_179&quot;, 
        &quot;Tk_180&quot;, &quot;Tk_181&quot;, &quot;Tk_182&quot;, &quot;Tk_183&quot;, 
        &quot;Tk_184&quot;, &quot;Tk_185&quot;, &quot;Tk_186&quot;, &quot;Tk_187&quot;, 
        &quot;Tk_188&quot;, &quot;Tk_189&quot;, &quot;Tk_19&quot;, &quot;Tk_190&quot;, 
        &quot;Tk_191&quot;, &quot;Tk_192&quot;, &quot;Tk_193&quot;, &quot;Tk_194&quot;, 
        &quot;Tk_195&quot;, &quot;Tk_196&quot;, &quot;Tk_197&quot;, &quot;Tk_198&quot;, 
        &quot;Tk_199&quot;, &quot;Tk_20&quot;, &quot;Tk_200&quot;, &quot;Tk_201&quot;, 
        &quot;Tk_202&quot;, &quot;Tk_203&quot;, &quot;Tk_204&quot;, &quot;Tk_205&quot;, 
        &quot;Tk_206&quot;, &quot;Tk_207&quot;, &quot;Tk_208&quot;, &quot;Tk_209&quot;, 
        &quot;Tk_210&quot;, &quot;Tk_211&quot;, &quot;Tk_212&quot;, &quot;Tk_213&quot;, 
        &quot;Tk_214&quot;, &quot;Tk_215&quot;, &quot;Tk_216&quot;, &quot;Tk_217&quot;, 
        &quot;Tk_218&quot;, &quot;Tk_219&quot;, &quot;Tk_220&quot;, &quot;Tk_221&quot;, 
        &quot;Tk_222&quot;, &quot;Tk_223&quot;, &quot;Tk_224&quot;, &quot;Tk_225&quot;, 
        &quot;Tk_27&quot;, &quot;Tk_3&quot;, &quot;Tk_34&quot;, &quot;Tk_38&quot;, 
        &quot;Tk_4&quot;, &quot;Tk_40&quot;, &quot;Tk_46&quot;, &quot;Tk_47&quot;, 
        &quot;Tk_48&quot;, &quot;Tk_49&quot;, &quot;Tk_50&quot;, &quot;Tk_51&quot;, 
        &quot;Tk_52&quot;, &quot;Tk_53&quot;, &quot;Tk_54&quot;, &quot;Tk_56&quot;, 
        &quot;Tk_57&quot;, &quot;Tk_59&quot;, &quot;Tk_60&quot;, &quot;Tk_61&quot;, 
        &quot;Tk_62&quot;, &quot;Tk_64&quot;, &quot;Tk_68&quot;, &quot;Tk_71&quot;, 
        &quot;Tk_72&quot;, &quot;Tk_73&quot;, &quot;Tk_74&quot;, &quot;Tk_76&quot;, 
        &quot;Tk_85&quot;, &quot;Tk_86&quot;, &quot;Tk_89&quot;, &quot;Tk_90&quot;, 
        &quot;Tk_91&quot;, &quot;Tk_95&quot;, &quot;Tk_96&quot;, &quot;Tk_97&quot;, 
        &quot;Tk_98&quot;, &quot;Tk_83&quot;, &quot;Tk_2&quot;, &quot;Tk_18&quot;, 
        &quot;Tk_35&quot;, &quot;Tk_84&quot;, &quot;Tk_37&quot;, &quot;Tk_5&quot;, 
        &quot;Tk_22&quot;, &quot;Tk_39&quot;, &quot;Tk_63&quot;, &quot;Tk_75&quot;, 
        &quot;Tk_87&quot;, &quot;Tk_99&quot;, &quot;Tk_6&quot;, &quot;Tk_24&quot;, 
        &quot;Tk_88&quot;, &quot;Tk_8&quot;, &quot;Tk_26&quot;, &quot;Tk_41&quot;, 
        &quot;Tk_65&quot;, &quot;Tk_77&quot;, &quot;Tk_9&quot;, &quot;Tk_42&quot;, 
        &quot;Tk_66&quot;, &quot;Tk_78&quot;, &quot;Tk_11&quot;, &quot;Tk_29&quot;, 
        &quot;Tk_43&quot;, &quot;Tk_55&quot;, &quot;Tk_79&quot;, &quot;Tk_12&quot;, 
        &quot;Tk_31&quot;, &quot;Tk_44&quot;, &quot;Tk_67&quot;, &quot;Tk_80&quot;, 
        &quot;Tk_92&quot;, &quot;Tk_104&quot;, &quot;Tk_13&quot;, &quot;Tk_32&quot;, 
        &quot;Tk_45&quot;, &quot;Tk_69&quot;, &quot;Tk_81&quot;, &quot;Tk_93&quot;, 
        &quot;Tk_105&quot;, &quot;Tk_15&quot;, &quot;Tk_33&quot;, &quot;Tk_58&quot;, 
        &quot;Tk_70&quot;, &quot;Tk_82&quot;, &quot;Tk_94&quot;, &quot;Tk_ntc14&quot;, 
        &quot;Tk_106&quot;, &quot;Tk_109&quot;, &quot;Tk_110&quot;, &quot;Tk_111&quot;, 
        &quot;Tk_112&quot;, &quot;Tk_113&quot;, &quot;Tk_114&quot;, &quot;Tk_115&quot;, 
        &quot;Tk_116&quot;, &quot;Tk_117&quot;, &quot;Tk_118&quot;, &quot;Tk_121&quot;, 
        &quot;Tk_123&quot;, &quot;Tk_124&quot;, &quot;Tk_125&quot;, &quot;Tk_126&quot;, 
        &quot;Tk_127&quot;, &quot;Tk_128&quot;, &quot;Tk_129&quot;, &quot;Tk_130&quot;, 
        &quot;Tk_132&quot;, &quot;Tk_133&quot;, &quot;Tk_135&quot;, &quot;Tk_137&quot;, 
        &quot;Tk_138&quot;, &quot;Tk_139&quot;, &quot;Tk_140&quot;, &quot;Tk_142&quot;, 
        &quot;Tk_145&quot;, &quot;Tk_146&quot;, &quot;Tk_147&quot;, &quot;Tk_Ecol1&quot;, 
        &quot;Tk_Ecol10&quot;, &quot;Tk_Ecol11&quot;, &quot;Tk_Ecol12&quot;, 
        &quot;Tk_Ecol13&quot;, &quot;Tk_Ecol14&quot;, &quot;Tk_Ecol15&quot;, 
        &quot;Tk_Ecol16&quot;, &quot;Tk_Ecol17&quot;, &quot;Tk_Ecol18&quot;, 
        &quot;Tk_Ecol19&quot;, &quot;Tk_Ecol2&quot;, &quot;Tk_Ecol20&quot;, 
        &quot;Tk_Ecol21&quot;, &quot;Tk_Ecol22&quot;, &quot;Tk_Ecol23&quot;, 
        &quot;Tk_Ecol24&quot;, &quot;Tk_Ecol25&quot;, &quot;Tk_Ecol3&quot;, 
        &quot;Tk_Ecol4&quot;, &quot;Tk_Ecol5&quot;, &quot;Tk_Ecol6&quot;, 
        &quot;Tk_Ecol7&quot;, &quot;Tk_Ecol8&quot;, &quot;Tk_Ecol9&quot;, 
        &quot;Tk_ntc15&quot;, &quot;Tk_ntc16&quot;, &quot;Tk_ntcecol1&quot;, 
        &quot;Tk_ntcecol2&quot;))))
###Not using tsv import anymore
#metadata &lt;- read_tsv(&quot;../data/dada2/phyloseq/metadata.tsv&quot;)

# Remove second row which contains column data for QIIME2 format
metadata &lt;- metadata[-c(1), ]
metadata_lab = column_to_rownames(metadata, var = &quot;sample-id&quot;)
sampledata = sample_data(data.frame(metadata_lab))
sampledata</code></pre>
<p><strong>Create final phyloseq object</strong></p>
<p>Now you can merge your data to create a final phyloseq object</p>
<pre class="r"><code>ps_raw_bact = merge_phyloseq(physeq, sampledata, rep.seqs)</code></pre>
<p>Remove samples with NA values or not part of final data set,</p>
<pre class="r"><code>ps_raw_bact &lt;- subset_samples(ps_raw_bact, !SampleType==&quot;EcolSample&quot;)</code></pre>
<p>An easy way to view the tables is using Nice Tables</p>
<pre class="r"><code>Nice.Table(ps_raw_bact@sam_data)</code></pre>
<pre class="r"><code>Nice.Table(ps_bact_samp@tax_table)</code></pre>
</div>
<div id="decontam" class="section level2">
<h2>2. Decontam</h2>
<p>R package <code>decontam</code> to assess contaminating OTUs, <a href="https://bioconductor.org/packages/release/bioc/vignettes/decontam/inst/doc/decontam_intro.html">tutorial</a>.</p>
<p>The CRAN version only works on R version &lt;4. To install for R versions &gt;4 install from <a href="https://bioconductor.org/packages/release/bioc/html/decontam.html">bioconductor</a> using the following</p>
<p>Make plot of library size of Samples vs Controls</p>
<pre class="r"><code>df &lt;- as.data.frame(sample_data(ps_raw_bact)) # Put sample_data into a ggplot-friendly data.frame
df$LibrarySize &lt;- sample_sums(ps_raw_bact)
df &lt;- df[order(df$LibrarySize),]
df$Index &lt;- seq(nrow(df))
libQC &lt;- ggplot(data=df, aes(x=Index, y=LibrarySize, color=SampleType)) + geom_point()
ggsave(&quot;libQC.pdf&quot;, plot = libQC, path = &quot;output/plots&quot;, width = 10, height = 10, units = &quot;cm&quot;)</code></pre>
<p>Make html plot with <code>plotly</code></p>
<pre class="r"><code>libQCplotly &lt;- ggplotly(libQC)
htmlwidgets::saveWidget(libQCplotly, &quot;output/plots/libQCplotly.html&quot;)</code></pre>
<p>Identify contaminating ASVs - define control samples and threshold (e.g. <code>0.05</code>)</p>
<pre class="r"><code>df &lt;- as.data.frame(sample_data(ps_raw_bact)) # Put sample_data into a ggplot-friendly data.frame
sample_data(ps_raw_bact)$is.neg &lt;- sample_data(ps_raw_bact)$SampleType == &quot;Control&quot;
contamdf.prev &lt;- isContaminant(ps_raw_bact, method=&quot;prevalence&quot;, neg=&quot;is.neg&quot;, threshold = 0.05)</code></pre>
<p>Identify contaminants</p>
<pre class="r"><code>table(contamdf.prev$contaminant)
head(which(contamdf.prev$contaminant))
table(contamdf.prev$contaminant)

con_ASVs &lt;- contamdf.prev %&gt;% 
  filter(contaminant == &quot;TRUE&quot;)
con_ASVs  &lt;- rownames(con_ASVs)</code></pre>
<p>Take a look at the number of times several of these taxa were observed in negative controls and positive samples.</p>
<pre class="r"><code># Make phyloseq object of presence-absence in negative controls and true samples
ps.pa &lt;- transform_sample_counts(ps_raw_bact, function(abund) 1*(abund&gt;0))
ps.pa.neg &lt;- prune_samples(sample_data(ps_raw_bact)$SampleType == &quot;Control&quot;, ps.pa)
ps.pa.pos &lt;- prune_samples(sample_data(ps.pa)$SampleType == &quot;Sample&quot;, ps.pa)
# Make data.frame of prevalence in positive and negative samples
df.pa &lt;- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.prev$contaminant)
# Make plot and save as pdf
deconplot &lt;- ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab(&quot;Prevalence (Negative Controls)&quot;) + ylab(&quot;Prevalence (True Samples)&quot;)
ggsave(&quot;deconplot.pdf&quot;, plot = deconplot, path = &quot;output/plots&quot;, width = 10, height = 10, units = &quot;cm&quot;)</code></pre>
<p>Make distribution plot of reads using <a href="https://microsud.github.io/microbiomeutilities/articles/microbiomeutilities.html#distribution-of-reads-1">microbiomeutilities</a></p>
<pre class="r"><code>distrib &lt;- plot_read_distribution(ps_raw_bact, groups = &quot;SampleCategory&quot;, 
                            plot.type = &quot;density&quot;) + xlab(&quot;Reads per sample&quot;) + ylab(&quot;Density&quot;)
distrib &lt;- distrib + geom_density(alpha = 0.5, fill = &quot;grey&quot;)
ggsave(&quot;distrib.pdf&quot;, plot = distrib, path = &quot;output/plots&quot;, width = 15, height = 10, units = &quot;cm&quot;)</code></pre>
</div>
<div id="subset-and-filter" class="section level2">
<h2>3. Subset and Filter</h2>
<p>As determined by <code>decontam</code> methods identify contaminant ASVs and threshold for positive reads. Then transform otu count data.</p>
<pre class="r"><code>ps_dec_bact &lt;- prune_taxa(!contamdf.prev$contaminant, ps_raw_bact)
ps_dec_bact@otu_table [, 1:702][ps_dec_bact@otu_table [, 1:702] &lt; 100] &lt;- 0</code></pre>
<p>Save R data for phyloseq object - saving “raw data” (<code>ps_raw_bact</code>) and “decontaminated data” (<code>ps_dec_bact</code>)</p>
<pre class="r"><code>save(ps_raw_bact, file = &quot;../data/dada2/phyloseq/ps_raw_bact.RData&quot;)
save(ps_dec_bact, file = &quot;../data/dada2/phyloseq/ps_dec_bact.RData&quot;)</code></pre>
<p>To load raw and decon data quickly from <code>.RData</code> format.</p>
<pre class="r"><code>load(&quot;../data/dada2/phyloseq/ps_raw_bact.RData&quot;)
load(&quot;../data/dada2/phyloseq/ps_dec_bact.RData&quot;)</code></pre>
<div id="for-phyloseq-object" class="section level3">
<h3>3.1. For phyloseq object</h3>
<p>Subset phyloseq object based on sample types</p>
<pre class="r"><code># Exclude controls
ps_bact_samp = subset_samples(ps_dec_bact, !SampleType==&quot;Control&quot;)
# Blood samples only
ps_bact_bl = subset_samples(ps_bact_samp, SampleCategory==&quot;Blood&quot;)
# Tissue samples only
ps_bact_tis = subset_samples(ps_bact_samp, SampleCategory==&quot;Tissue&quot;)
# Tick samples only
ps_bact_tick = subset_samples(ps_bact_samp, SampleCategory==&quot;Tick&quot;)</code></pre>
<p>Subset certain taxa of interest</p>
<pre class="r"><code># taxa of interest
ps_bact_samp_sub = subset_taxa(ps_bact_samp, Order==&quot;Coxiellales&quot; | Order==&quot;Mycoplasmataceae&quot; | Genus==&quot;Bartonella&quot; | Order==&quot;Francisellales&quot; | Genus==&quot;Borreliella&quot; | Genus==&quot;Leptospira&quot; | Family==&quot;Rickettsiaceae&quot; | Family==&quot;Anaplasmataceae&quot; | Family==&quot;Midichloriaceae&quot; | Genus==&quot;Rickettsiella&quot;)
ps_bact_samp_sub = prune_samples(sample_sums(ps_bact_samp_sub)&gt;=1000, ps_bact_samp_sub)
ps_bact_samp_sub = filter_taxa(ps_bact_samp_sub, function(x) sum(x &gt; 1) &gt; (0.01*length(x)), TRUE)</code></pre>
<p>Add tree for this taxa</p>
<pre class="r"><code>tree2 &lt;- read_tree(&quot;data/dada2/top38.nwk&quot;)
ps_bact_samp_sub &lt;- merge_phyloseq(ps_bact_samp_sub, tree)</code></pre>
<p>Subset phyloseq object based on host species</p>
<pre class="r"><code># Black rat
ps_BR = subset_samples(ps_bact_samp, species==&quot;Black rat&quot;)
# Brush tail possum
ps_BTP = subset_samples(ps_bact_samp, species==&quot;Brush tail possum&quot;)
# Chuditch
ps_chud = subset_samples(ps_bact_samp, species==&quot;Chuditch&quot;)
# Long-nosed bandicoot
ps_LNB = subset_samples(ps_bact_samp, species==&quot;Long-nosed bandicoot&quot;)</code></pre>
</div>
<div id="for-ampvis2-object" class="section level3">
<h3>3.2. For ampvis2 object</h3>
<p><strong>Make ampvis2 object for analysis</strong></p>
<pre class="r"><code>#require the devtools package to source gist
if(!require(&quot;devtools&quot;))
  install.packages(&quot;devtools&quot;)
#source the phyloseq_to_ampvis2() function from the gist
devtools::source_gist(&quot;8d0ca4206a66be7ff6d76fc4ab8e66c6&quot;)

#convert
bact_amp &lt;- phyloseq_to_ampvis2(ps_dec_bact)</code></pre>
<p>Subset ampvis2 object based on sample category</p>
<pre class="r"><code>#remove controls
amp_samp &lt;- amp_subset_samples(bact_amp, 
                                 !SampleType %in% c(&quot;Control&quot;),
                                 RemoveAbsents = TRUE)
#blood samples
amp_bl &lt;- amp_subset_samples(amp_samp, 
                                 SampleCategory %in% c(&quot;Blood&quot;),
                                 RemoveAbsents = TRUE)
#tissue samples
am_tis &lt;- amp_subset_samples(amp_samp, 
                                 SampleCategory %in% c(&quot;Tissue&quot;),
                                 RemoveAbsents = TRUE)
#tick samples
amp_tick &lt;- amp_subset_samples(amp_samp, 
                                 SampleCategory %in% c(&quot;Tick&quot;),
                                 RemoveAbsents = TRUE)
#tick samples
amp_tick_sub &lt;- amp_subset_samples(amp_tick, 
                                 instar %in% c(&quot;Female&quot;, &quot;Larvae&quot;, &quot;Nymph&quot;),
                                 RemoveAbsents = TRUE)</code></pre>
<p>Ampvis2 have an easy way to merge replicate samples (i.e. duplicate PCR amplicons)</p>
<pre class="r"><code>#merge by gDNAID
dmerged &lt;- amp_mergereplicates(ps_bact_samp,
  merge_var = &quot;gDNAID&quot;,
  round = &quot;up&quot;
)
dmerged$metadata</code></pre>
<p>Filter out low abundant samples</p>
<pre class="r"><code>amp_samp1000 &lt;- amp_subset_samples(amp_samp,
                                 minreads = 1000,
                                 RemoveAbsents = TRUE)
amp_samp1000</code></pre>
<p>Subset certain species and sites</p>
<pre class="r"><code>chud &lt;- amp_subset_samples(amp_samp, 
                              species %in% c(&quot;Chuditch&quot;),
                              !SampleCategory %in% c(&quot;Tick&quot;),
                              RemoveAbsents = TRUE)
LNB &lt;- amp_subset_samples(amp_samp, 
                              species %in% c(&quot;Long-nosed bandicoot&quot;),
                              RemoveAbsents = TRUE)
BTP &lt;- amp_subset_samples(amp_samp, 
                              species %in% c(&quot;Brush tail possum&quot;),
                              RemoveAbsents = TRUE)
BR &lt;- amp_subset_samples(amp_samp, 
                              species %in% c(&quot;Black rat&quot;),
                              RemoveAbsents = TRUE)</code></pre>
<blockquote>
<p>Set colours when comparing blood, tick and tissue samples.</p>
</blockquote>
<pre class="r"><code>ColSampCat = c(&quot;#7a255d&quot;, &quot;#9fd0cb&quot;, &quot;#7566ff&quot;)</code></pre>
</div>
</div>
<div id="explore-data" class="section level2">
<h2>4. Explore data</h2>
<p>Subset phyloseq object and convert to ‘long’ dataframe to quickly identify samples.</p>
<p>Identify unidentified taxa at kingdom level</p>
<pre class="r"><code># subset taxa to include those with no Kingdom assignment
kingdom_na = subset_taxa(ps_bact_samp, is.na(Kingdom))
# remove taxa with 0 abundance
kingdom_na &lt;- prune_taxa(taxa_sums(kingdom_na) &gt; 0, kingdom_na)
# write sequences to fasta file
writeXStringSet(kingdom_na@refseq, &quot;data/dada2/unclassified_taxa/kingdom_na.fa&quot;)
# melt dataframe into long df
m_kingdom = psmelt(kingdom_na)
#Remove all rows were abundance is zero
m_kingdom2 = m_kingdom[m_kingdom$Abundance != 0, ]
# write melt data frame to csv file
write.csv(m_kingdom2, &quot;data/dada2/unclassified_taxa/kingdom_na.csv&quot;)</code></pre>
<p>Taxa of interest - example</p>
<pre class="r"><code># subset taxa to include those with no Kingdom assignment
bart = subset_taxa(ps_bact_samp, Genus==&quot;Rickettsia&quot;)
# remove taxa with 0 abundance
bart = prune_taxa(taxa_sums(bart) &gt; 0, bart)
# melt dataframe into long df
m_bart= psmelt(bart)
#Remove all rows were abundance is zero
m_bart2 = m_bart[m_bart$Abundance != 0, ]

# identify number of sample positive (i.e. `gDNAID``)
n_distinct(m_bart2$gDNAID)
# identify number of animals positive (i.e. `animalID``)
n_distinct(m_bart2$animalID)
unique(m_bart2$species)

tmp = m_bart2 %&gt;%
  filter(SampleCategory == &quot;Blood&quot;)
# identify number of sample positive (i.e. `gDNAID``)
n_distinct(tmp$gDNAID)
# identify number of animals positive (i.e. `animalID``)
n_distinct(tmp$animalID)
# List host species
unique(tmp$species)
unique(tmp$tick_species)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
