<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>QIIME2</title>

<script src="site_libs/header-attrs-2.7.2/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">wildlife-bacteria</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="site-map.html">Sites</a>
</li>
<li>
  <a href="QIIME2.html">QIIME2</a>
</li>
<li>
  <a href="phyloseq.html">Phyloseq</a>
</li>
<li>
  <a href="microbiome-viz.html">Microbiome viz</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/siobhon-egan/wildlife-bacteria">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">QIIME2</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>wildlife-bacteria/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsiobhoneganwildlifebacteriatree6ebcc5da3c0fae6774b808086ce0c4cf4b9c3fb4targetblank6ebcc5da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/siobhon-egan/wildlife-bacteria/tree/6ebcc5da3c0fae6774b808086ce0c4cf4b9c3fb4" target="_blank">6ebcc5d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsiobhoneganwildlifebacteriatree6ebcc5da3c0fae6774b808086ce0c4cf4b9c3fb4targetblank6ebcc5da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/siobhon-egan/wildlife-bacteria/tree/6ebcc5da3c0fae6774b808086ce0c4cf4b9c3fb4" target="_blank">6ebcc5d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/plots/.DS_Store
    Ignored:    output/plots/QC/.DS_Store
    Ignored:    output/plots/boxplots_select_taxa/.DS_Store
    Ignored:    output/plots/heatmaps/.DS_Store
    Ignored:    output/plots/maps/.DS_Store
    Ignored:    output/plots/tax_prev_abund/.DS_Store

Untracked files:
    Untracked:  NCBI_data/
    Untracked:  analysis/microbiome-viz-extra.Rmd
    Untracked:  analysis/phylogeny.Rmd
    Untracked:  analysis/tois.Rmd
    Untracked:  data/dada2/
    Untracked:  data/dada2_tois/
    Untracked:  data/taxa_trees/
    Untracked:  data/tmp/
    Untracked:  output/beta-div-statistics.txt
    Untracked:  output/supp_table_pos.xlsx
    Untracked:  tmp/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/QIIME2.Rmd</code>) and HTML (<code>docs/QIIME2.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/siobhon-egan/wildlife-bacteria/blob/6ebcc5da3c0fae6774b808086ce0c4cf4b9c3fb4/analysis/QIIME2.Rmd" target="_blank">6ebcc5d</a>
</td>
<td>
siobhon-egan
</td>
<td>
2021-04-24
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/siobhon-egan/wildlife-bacteria/6ebcc5da3c0fae6774b808086ce0c4cf4b9c3fb4/docs/QIIME2.html" target="_blank">6ebcc5d</a>
</td>
<td>
siobhon-egan
</td>
<td>
2021-04-24
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/siobhon-egan/wildlife-bacteria/5486f7e4f4a6cfe483395735ecfa6254cd45174b/docs/QIIME2.html" target="_blank">5486f7e</a>
</td>
<td>
siobhon-egan
</td>
<td>
2021-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/siobhon-egan/wildlife-bacteria/blob/ebe20d01dc71c1ba2eb30c7ab053c4d2d0c78821/analysis/QIIME2.Rmd" target="_blank">ebe20d0</a>
</td>
<td>
siobhon-egan
</td>
<td>
2021-02-26
</td>
<td>
add qiime and phyloseq pages
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Analysis of 16S rRNA (hypervariable region v1-2) metabarcoding.</p>
<p>Raw Illumina MiSeq <code>.fastq.gz</code> reads analysed using QIIME2-2020.11 pipeline using dada2 denoising to create ASVs.</p>
<p>Author: Siobhon Egan <a href="mailto:siobhonegan@hotmail.com" class="email">siobhonegan@hotmail.com</a><a href="mailto:siobhon.egan@murdoch.edu.au" class="email">siobhon.egan@murdoch.edu.au</a> Last updated: Jan 20201</p>
<p><strong>Background</strong></p>
<p>This workflow is written for analysing amplicon data in QIIME2. Input data is Illumina MiSeq paired-end data prepared using Nextera XT indexes (i.e. no additional demultiplexing steps are needed in this case however should your data require demultiplexing it can easily be added in).</p>
<div id="install-activate-qiime2-environment-commandline" class="section level2">
<h2>0. Install &amp; activate QIIME2 environment (commandline)</h2>
<p>This workflow utilsing commandline interface with QIIME2.</p>
<p>Requires miniconda/conda, see <a href="https://docs.qiime2.org/2020.11/install/native/#installing-miniconda">here</a></p>
<p>Latest version = <a href="https://docs.qiime2.org/2020.11/install/native/">QIIME2-2020.11</a>, see QIIME2 documentation for install based on your platform.</p>
<p><strong>Activate qiime2 environment</strong></p>
<pre class="bash"><code>conda activate qiime2-2020.11</code></pre>
</div>
<div id="input-data" class="section level2">
<h2>1. Input data</h2>
<blockquote>
<p>Assumes paired-end data that does not require demultiplexing</p>
</blockquote>
<p>Place raw data files in zipped format (i.e. <code>.fastq.gz</code> in a directory called <code>raw_data/</code>).</p>
<div id="file-naming-conventions" class="section level3">
<h3>File naming conventions</h3>
<p>In <a href="https://docs.qiime2.org/2020.11/tutorials/importing/#casava-1-8-paired-end-demultiplexed-fastq">Casava 1.8 demultiplexed (paired-end)</a> format, there are two <code>.fastq.gz</code> files for each sample in the study, each containing the forward or reverse reads for that sample. The file name includes the sample identifier. The forward and reverse read file names for a single sample might look like <code>XXXX_L001_R1_001.fastq.gz</code> and <code>XXXX_L001_R2_001.fastq.gz</code>, respectively. The underscore-separated fields in this file name are:</p>
<ul>
<li>the sample identifier,</li>
<li>the barcode sequence or a barcode identifier,</li>
<li>the lane number,</li>
<li>the direction of the read (i.e. R1 or R2), and</li>
<li>the set number.</li>
</ul>
<p>Depending on sequencing facility you may need to add the <code>_001</code> prefix to sample files.</p>
<p>Note however that you do <strong>not</strong> need to unzip fastq data to analyse.</p>
<p>Navigate into the directory with raw data files:</p>
<pre class="bash"><code>for file in raw_data/*.fastq.gz;
do
newname=$(echo &quot;$file&quot; | sed &#39;s/0_BPDNR//&#39; | sed &#39;s/.fastq/_001.fastq/&#39;)
mv $file $newname
done</code></pre>
</div>
<div id="import-as-qiime2-artefact" class="section level3">
<h3>Import as QIIME2 artefact</h3>
<p>Import <code>.fastq.gz</code> data into QIIME2 format using <a href="https://docs.qiime2.org/2020.11/tutorials/importing/#casava-1-8-paired-end-demultiplexed-fastq">Casava 1.8 demultiplexed (paired-end)</a> option. Remember assumes raw data is in directory labelled <code>raw_data/</code> and file naming format as above.</p>
<pre class="bash"><code>qiime tools import \
--type &#39;SampleData[PairedEndSequencesWithQuality]&#39; \
--input-path raw_data \
--input-format CasavaOneEightSingleLanePerSampleDirFmt \
--output-path 16S_demux_seqs.qza</code></pre>
<p>In this case we are using Nextera Indexes which mean they are demultiplexed automatically by basespace and therefore we can skip over any reference to demultiplexing steps.</p>
<p><strong>Inspect reads for quality</strong> To inspect raw reads</p>
<pre class="bash"><code>qiime demux summarize \
  --i-data 16S_demux_seqs.qza \
  --o-visualization 16S_demux_seqs.qzv</code></pre>
<p>View this output by importing into <a href="https://view.qiime2.org/">QIIME2 view</a>. Use this output to choose your parameters for QC such as trimming low quality sequences and truncating sequence length.</p>
</div>
<div id="sample-metadata" class="section level3">
<h3>Sample metadata</h3>
<p>This holds you associated metadata related to your samples (e.g. host information, sampling data, etc). <a href="https://docs.qiime2.org/2020.11/tutorials/moving-pictures/#sample-metadata">Tutorial here</a></p>
<p>The metadata needs to be in <code>.tsv</code> format, the best way to do this is to access the QIIME2 googlesheet example. Save a copy and edit/add in your sample details. Then select <code>File &gt; Download as &gt; Tab-separated values</code>. Alternatively, the command <code>wget "https://data.qiime2.org/2020.11/tutorials/moving-pictures/sample_metadata.tsv"</code> will download the sample metadata as tab-separated text and save it in the file <code>sample-metadata.tsv</code>. It is import you don’t change the header for the first column <code>sample-id</code>.</p>
</div>
</div>
<div id="sequence-quality-control-and-feature-table-construction" class="section level2">
<h2>2. Sequence quality control and feature table construction</h2>
<blockquote>
<p>Denoise using dada2</p>
</blockquote>
<p>Based on quality plot in the above output <code>16S_demux_seqs.qza</code> adjust trim length to where quality falls.</p>
<p>Then you can also trim primers. In this case working with 16S v1-2 data with the following primers</p>
<p>Example data - amplicon NGS data targeting bacteria using 16S rRNA hypervariable region 1-2 with the following primers:</p>
<ul>
<li>27F-Y (20 nt): AGAGTTTGATCCTGGCTYAG #16S v1-2 primer, ref Gofton et al. Parasites &amp; Vectors (2015) 8:345</li>
<li>338R (19 nt): TGCTGCCTCCCGTAGGAGT #16S v1-2 primer, ref Turner et al. J Eukaryot Microbiol (1999) 46(4):32</li>
</ul>
<pre class="bash"><code>qiime dada2 denoise-paired \
  --i-demultiplexed-seqs 16S_demux_seqs.qza \
  --p-trim-left-f 20 \
  --p-trim-left-r 19 \
  --p-trunc-len-f 250 \
  --p-trunc-len-r 250 \
  --o-table 16S_denoise_table.qza \
  --o-representative-sequences 16S_denoise_rep-seqs.qza \
  --o-denoising-stats 16S_denoise-stats.qza</code></pre>
<p>At this stage, you will have artifacts containing the feature table, corresponding feature sequences, and DADA2 denoising stats. You can generate summaries of these as follows.</p>
<pre class="bash"><code>qiime feature-table summarize \
  --i-table 16S_denoise_table.qza \
  --o-visualization 16S_denoise_table.qzv \
  --m-sample-metadata-file sample-metadata.tsv # Can skip this bit if needed.

qiime feature-table tabulate-seqs \
  --i-data 16S_denoise_rep-seqs.qza \
  --o-visualization 16S_denoise_rep-seqs.qzv

qiime metadata tabulate \
  --m-input-file 16S_denoise-stats.qza \
  --o-visualization 16S_denoise-stats.qzv</code></pre>
<div id="merging-denoised-artefacts" class="section level3">
<h3>Merging denoised artefacts</h3>
<p>To merge denoised data sets and generate one <code>FeatureTable[Frequency]</code> and <code>FeatureData[Sequence]</code> artifacts</p>
<pre class="bash"><code>qiime feature-table merge \
  --i-tables table-1.qza \
  --i-tables table-2.qza \
  --o-merged-table table.qza
qiime feature-table merge-seqs \
  --i-data rep-seqs-1.qza \
  --i-data rep-seqs-2.qza \
  --o-merged-data rep-seqs.qza</code></pre>
</div>
<div id="export-asv-table" class="section level3">
<h3>Export ASV table</h3>
<p>To produce an ASV table with number of each ASV reads per sample that you can open in excel. Use tutorial <a href="https://rstudio-pubs-static.s3.amazonaws.com/489645_5fff8a6a02d84084a55e3b5b6ff960a4.html#">here</a></p>
<p>Need to make biom file first</p>
<pre class="bash"><code>qiime tools export \
--input-path 16S_denoise_table.qza \
--output-path feature-table

biom convert \
-i feature-table/feature-table.biom \
-o feature-table/feature-table.tsv \
--to-tsv</code></pre>
</div>
<div id="phylogenetic-tree" class="section level3">
<h3>Phylogenetic tree</h3>
<p>Several downstream diversity metrics, available within QIIME 2, require that a phylogenetic tree be constructed using the Operational Taxonomic Units (OTUs) or Amplicon Sequence Variants (ASVs) being investigated. Documentation <a href="https://docs.qiime2.org/2020.11/tutorials/phylogeny/">here</a></p>
<pre class="bash"><code>qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences rep-seqs.qza \
  --o-alignment aligned-rep-seqs.qza \
  --o-masked-alignment masked-aligned-rep-seqs.qza \
  --o-tree unrooted-tree.qza \
  --o-rooted-tree rooted-tree.qza</code></pre>
<p><strong>Export</strong></p>
<p>Covert unrooted tree output to newick formatted file</p>
<pre class="bash"><code>qiime tools export \
  --input-path unrooted-tree.qza \
  --output-path exported-tree</code></pre>
</div>
</div>
<div id="taxonomy" class="section level2">
<h2>3.Taxonomy</h2>
<p>Assign taxonomy to denoised sequences using a pre-tarined naive bayes classifier and the <a href="https://docs.qiime2.org/2020.11/plugins/available/feature-classifier/">q2-feature-classifier</a> plugin. Details on how to create a classifier are available <a href="link">here</a>.</p>
<p>Note that taxonomic classifiers perform best when they are trained based on your specific sample preparation and sequencing parameters, including the primers that were used for amplification and the length of your sequence reads.</p>
<pre class="bash"><code>qiime feature-classifier classify-sklearn \
--i-classifier /Taxonomy/QIIME2_classifiers_v2020.11/Silva_99_Otus/27F-388Y/classifier.qza \
--i-reads 16S_denoise_rep-seqs.qza \
--o-classification qiime2-taxa-silva/taxonomy.qza

qiime metadata tabulate \
--m-input-file qiime2-taxa-silva/taxonomy.qza \
--o-visualization qiime2-taxa-silva/taxonomy.qzv</code></pre>
<p>In order to be able to download the sample OTU table need to do the taxonomy assignment and then make the taxa barplot. Then can download csv file with sequence number, samples and taxonomy. <a href="https://docs.qiime2.org/2020.11/tutorials/moving-pictures/#taxonomic-analysis">see here</a></p>
<pre><code>qiime taxa barplot \
  --i-table table.qza \
  --i-taxonomy taxonomy.qza \
  --m-metadata-file sample-metadata.tsv \
  --o-visualization taxa-bar-plots.qzv</code></pre>
<p>Details on sample metadata available <a href="https://docs.qiime2.org/2020.11/tutorials/moving-pictures/#sample-metadata">here</a></p>
<p>Extra bit of code to generate a taxonomy table table to tsv from the commandline</p>
<pre class="bash"><code>qiime tools export \
--input-path taxonomy.qza \
--output-path exports</code></pre>
<hr />
</div>
<div id="extra-info" class="section level2">
<h2>Extra info</h2>
<p>Place to leave some links</p>
<ul>
<li><a href="https://rstudio-pubs-static.s3.amazonaws.com/489645_5fff8a6a02d84084a55e3b5b6ff960a4.html#">QIIME2 Introductory Workshop by Robert W. Murdoch</a></li>
<li><a href="https://github.com/jbisanz/qiime2R">Tutorial: Integrating QIIME2 and R for data visualization and analysis using qiime2R (v0.99.4)</a></li>
<li><a href="https://joey711.github.io/phyloseq/index.html">Phyloseq webpage</a></li>
<li><a href="https://vaulot.github.io/tutorials/Phyloseq_tutorial.html#content">Phyloseq tutorial</a></li>
<li><a href="http://14.139.189.27/ngsdat19/downloads/NGSDAT_2019_TRANING%20MANUAL.pdf">Bioinformatics for metagenome data analysis by Nirmal Babu</a></li>
<li><a href="https://github.com/shu251/qiime2_ASVworkflow_v8/blob/master/README.md">QIIME2 workflow - determining Amplicon Sequence Variants by Sarah Hu</a></li>
<li><a href="https://micca.readthedocs.io/en/latest/index.html">micca (MICrobial Community Analysis)</a></li>
<li><a href="https://www.nicholas-ollberding.com/post/denoising-amplicon-sequence-variants-using-dada2-deblur-and-unoise3-with-qiime2/">Denoising Amplicon Sequence Variants Using DADA2, DeBlur, and UNOISE3 with QIIME2 by Nicholas Ollberding</a></li>
<li><a href="https://bioinformaticsworkbook.org/dataAnalysis/Metagenomics/Qiime2.html#gsc.tab=0">Amplicon analysis with QIIME2 Adam Rivers</a></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
