<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Taxa of interest</title>

<script src="site_libs/header-attrs-2.7.2/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">wildlife-bacteria</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="site-map.html">Sites</a>
</li>
<li>
  <a href="QIIME2.html">QIIME2</a>
</li>
<li>
  <a href="phyloseq.html">Phyloseq</a>
</li>
<li>
  <a href="microbiome-viz.html">Microbiome viz</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/siobhon-egan/wildlife-bacteria">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Taxa of interest</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>wildlife-bacteria/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210129code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210129code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210129)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsiobhoneganwildlifebacteriatreea0173d49bf2a7ffe8bebaf994dedf7048daaa07ctargetblanka0173d4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/siobhon-egan/wildlife-bacteria/tree/a0173d49bf2a7ffe8bebaf994dedf7048daaa07c" target="_blank">a0173d4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsiobhoneganwildlifebacteriatreea0173d49bf2a7ffe8bebaf994dedf7048daaa07ctargetblanka0173d4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/siobhon-egan/wildlife-bacteria/tree/a0173d49bf2a7ffe8bebaf994dedf7048daaa07c" target="_blank">a0173d4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  NCBI_data/
    Untracked:  RData/
    Untracked:  analysis/microbiome-viz-extra.Rmd
    Untracked:  analysis/phylogeny.Rmd
    Untracked:  analysis/tois.Rmd
    Untracked:  data/dada2/
    Untracked:  data/dada2_tois/
    Untracked:  data/ps_raw_bact.RData
    Untracked:  data/site_map.xlsx
    Untracked:  data/taxa_of_interest.RData
    Untracked:  data/taxa_trees/
    Untracked:  data/tmp/
    Untracked:  output/beta-div-statistics.txt
    Untracked:  output/plots/
    Untracked:  output/supp_table_pos.xlsx
    Untracked:  tmp/

Unstaged changes:
    Modified:   analysis/QIIME2.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/microbiome-viz.Rmd
    Modified:   analysis/phyloseq.Rmd

Staged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/microbiome-viz.Rmd
    Modified:   analysis/phyloseq.Rmd
    New:        analysis/site-map.Rmd
    New:        code/taxa_summary.R
    New:        data/ps_bact_samp.RData
    New:        data/ps_dec_bact.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level2">
<h2>0. Load libraries</h2>
<p><strong>Install libraries if required</strong></p>
<p>Only need to run this code once.</p>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
# phyloseq
source(&#39;http://bioconductor.org/biocLite.R&#39;)
biocLite(&#39;phyloseq&#39;)
#tidyverse
install. packages(&quot;tidyverse&quot;)
#ampvis2
install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;MadsAlbertsen/ampvis2&quot;)
#ampvis2extras
install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;kasperskytte/ampvis2extras&quot;)
#ggpubr
install.packages(&quot;ggpubr&quot;)
#agricolae
install.packages(&quot;agricolae&quot;)
install.packages(&quot;remotes&quot;)
remotes::install_github(&quot;DanielSprockett/reltools&quot;)
devtools::install_github(&#39;jsilve24/philr&#39;)
#decontam
BiocManager::install(&quot;decontam&quot;)
library(decontam)
if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;Biostrings&quot;)</code></pre>
<p><strong>Load libraries</strong></p>
<pre class="r"><code>pkgs &lt;- c(&quot;qiime2R&quot;, &quot;phyloseq&quot;, &quot;tidyverse&quot;, &quot;ampvis2&quot;, &quot;ampvis2extras&quot;, 
          &quot;ggpubr&quot;, &quot;agricolae&quot;, &quot;plotly&quot;, &quot;viridis&quot;, &quot;cowplot&quot;, &quot;MicrobeR&quot;, 
          &quot;microbiome&quot;, &quot;reshape&quot;, &quot;decontam&quot;, &quot;data.table&quot;, &quot;ape&quot;, &quot;DESeq2&quot;, 
          &quot;vegan&quot;, &quot;microbiomeutilities&quot;, &quot;knitr&quot;, &quot;tibble&quot;, &quot;dplyr&quot;, 
          &quot;patchwork&quot;, &quot;Biostrings&quot;)
lapply(pkgs, require, character.only = TRUE)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>1. Load data</h2>
<p>Quick load of raw and decon data (<code>.RData</code> format) as per <a href="phyloseq.html">phyloseq</a> workflow.</p>
<pre class="r"><code>load(&quot;RData/ps_raw_bact.RData&quot;)
load(&quot;RData/ps_dec_bact.RData&quot;)</code></pre>
</div>
<div id="subset" class="section level2">
<h2>2. Subset</h2>
<div id="sample-categoryies" class="section level3">
<h3>2.1 Sample Categoryies</h3>
<p>Subset phyloseq object based on sample types</p>
<pre class="r"><code># Exclude controls
ps_bact_samp = subset_samples(ps_dec_bact, !SampleType==&quot;Control&quot;)
# Blood samples only
ps_bact_bl = subset_samples(ps_bact_samp, SampleCategory==&quot;Blood&quot;)
# Tissue samples only
ps_bact_tis = subset_samples(ps_bact_samp, SampleCategory==&quot;Tissue&quot;)
# Tick samples only
ps_bact_tick = subset_samples(ps_bact_samp, SampleCategory==&quot;Tick&quot;)</code></pre>
</div>
<div id="unassigned" class="section level3">
<h3>2.2 Unassigned</h3>
<p>Filter ASVs for unassigned taxa and write fasta files and abundance tables</p>
<pre class="r"><code>## Unassigned Kingdom
kingdom_na = subset_taxa(ps_bact_samp, is.na(Kingdom))
kingdom_na &lt;- prune_taxa(taxa_sums(kingdom_na) &gt; 0, kingdom_na)
writeXStringSet(kingdom_na@refseq, &quot;data/dada2/unclassified_taxa/kingdom_na.fa&quot;)
# melt dataframe into long df
m_kingdom = psmelt(kingdom_na)
#Remove all rows were abundance is zero
m_kingdom2 = m_kingdom[m_kingdom$Abundance != 0, ]
write.csv(m_kingdom2, &quot;data/dada2/unclassified_taxa/kingdom_na.csv&quot;)

## Unassigned Phylum
phylum_na = subset_taxa(ps_bact_samp, is.na(Phylum))
phylum_na &lt;- prune_taxa(taxa_sums(phylum_na) &gt; 0, phylum_na)
writeXStringSet(phylum_na@refseq, &quot;data/dada2/unclassified_taxa/phylum_na.fa&quot;)
# melt dataframe into long df
m_phylum = psmelt(phylum_na)
#Remove all rows were abundance is zero
m_phylum2 = m_phylum[m_phylum$Abundance != 0, ]
write.csv(m_phylum2, &quot;data/dada2/unclassified_taxa/phylum_na.csv&quot;)

## Unassigned Class
class_na = subset_taxa(ps_bact_samp, is.na(Class))
class_na &lt;- prune_taxa(taxa_sums(class_na) &gt; 0, class_na)
writeXStringSet(class_na@refseq, &quot;data/dada2/unclassified_taxa/class_na.fa&quot;)
# melt dataframe into long df
m_class = psmelt(class_na)
#Remove all rows were abundance is zero
m_class2 = m_class[m_class$Abundance != 0, ]
write.csv(m_class2, &quot;data/dada2/unclassified_taxa/class_na.csv&quot;)

## Unassigned Order
order_na = subset_taxa(ps_bact_samp, is.na(Order))
order_na &lt;- prune_taxa(taxa_sums(order_na) &gt; 0, order_na)
writeXStringSet(order_na@refseq, &quot;data/dada2/unclassified_taxa/order_na.fa&quot;)
# melt dataframe into long df
m_order = psmelt(order_na)
#Remove all rows were abundance is zero
m_order2 = m_order[m_order$Abundance != 0, ]
write.csv(m_order2, &quot;data/dada2/unclassified_taxa/order_na.csv&quot;)

## Unassigned Family
family_na = subset_taxa(ps_bact_samp, is.na(Family))
family_na &lt;- prune_taxa(taxa_sums(family_na) &gt; 0, family_na)
writeXStringSet(family_na@refseq, &quot;data/dada2/unclassified_taxa/family_na.fa&quot;)
# melt dataframe into long df
m_family = psmelt(family_na)
#Remove all rows were abundance is zero
m_family2 = m_family[m_family$Abundance != 0, ]
write.csv(m_family2, &quot;data/dada2/unclassified_taxa/family_na.csv&quot;)</code></pre>
</div>
<div id="taxa-of-interest" class="section level3">
<h3>2.3 Taxa of interest</h3>
<p>Filter ASVs for taxa of interest and write as fasta file</p>
<pre class="r"><code>### Filter by taxa and then filter out ASVs with 0 sequences
Anaplasmataceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Anaplasmataceae&quot;)
Anaplasmataceae &lt;- prune_taxa(taxa_sums(Anaplasmataceae) &gt; 0, Anaplasmataceae)
writeXStringSet(Anaplasmataceae@refseq, &quot;data/dada2_tois/Anaplasmataceae.fa&quot;)
Rickettsiaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Rickettsiaceae&quot;)
Rickettsiaceae &lt;- prune_taxa(taxa_sums(Rickettsiaceae) &gt; 0, Rickettsiaceae)
writeXStringSet(Rickettsiaceae@refseq, &quot;data/dada2_tois/Rickettsiaceae.fa&quot;)
Midichloriaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Midichloriaceae&quot;)
Midichloriaceae &lt;- prune_taxa(taxa_sums(Midichloriaceae) &gt; 0, Midichloriaceae)
writeXStringSet(Midichloriaceae@refseq, &quot;data/dada2_tois/Midichloriaceae.fa&quot;)
Coxiellaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Coxiellaceae&quot;)
Coxiellaceae &lt;- prune_taxa(taxa_sums(Coxiellaceae) &gt; 0, Coxiellaceae)
writeXStringSet(Coxiellaceae@refseq, &quot;data/dada2_tois/Coxiellaceae.fa&quot;)
Mycoplasmataceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Mycoplasmataceae&quot;)
Mycoplasmataceae &lt;- prune_taxa(taxa_sums(Mycoplasmataceae) &gt; 0, Mycoplasmataceae)
writeXStringSet(Mycoplasmataceae@refseq, &quot;data/dada2_tois/Mycoplasmataceae.fa&quot;)
Francisellaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Francisellaceae&quot;)
Francisellaceae &lt;- prune_taxa(taxa_sums(Francisellaceae) &gt; 0, Francisellaceae)
writeXStringSet(Francisellaceae@refseq, &quot;data/dada2_tois/Francisellaceae.fa&quot;)
Diplorickettsiaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Diplorickettsiaceae&quot;)
Diplorickettsiaceae &lt;- prune_taxa(taxa_sums(Diplorickettsiaceae) &gt; 0, Diplorickettsiaceae)
writeXStringSet(Diplorickettsiaceae@refseq, &quot;data/dada2_tois/Diplorickettsiaceae.fa&quot;)
Rickettsiales &lt;- subset_taxa(ps_dec_bact, Order == &quot;Rickettsiales&quot;)
Rickettsiales &lt;- prune_taxa(taxa_sums(Rickettsiales) &gt; 0, Rickettsiales)
writeXStringSet(Rickettsiales@refseq, &quot;data/dada2_tois/Rickettsiales.fa&quot;)
Borrelia &lt;- subset_taxa(ps_dec_bact, Genus == &quot;Borrelia&quot;)
Borrelia &lt;- prune_taxa(taxa_sums(Borrelia) &gt; 0, Borrelia)
writeXStringSet(Borrelia@refseq, &quot;data/dada2_tois/Borrelia&quot;)
Bartonella &lt;- subset_taxa(ps_dec_bact, Genus == &quot;Bartonella&quot;)
Bartonella &lt;- prune_taxa(taxa_sums(Bartonella) &gt; 0, Bartonella)
writeXStringSet(Bartonella@refseq, &quot;data/dada2_tois/Bartonella.fa&quot;)
Eimeria &lt;- subset_taxa(ps_dec_bact, Family == &quot;Eimeria praecox&quot;)
Eimeria &lt;- prune_taxa(taxa_sums(Eimeria) &gt; 0, Eimeria)
writeXStringSet(Eimeria@refseq, &quot;data/dada2_tois/Eimeria.fa&quot;)
Mycobacteriaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Mycobacteriaceae&quot;)
Mycobacteriaceae &lt;- prune_taxa(taxa_sums(Mycobacteriaceae) &gt; 0, Mycobacteriaceae)
writeXStringSet(Mycobacteriaceae@refseq, &quot;data/dada2_tois/Mycobacteriaceae.fa&quot;)</code></pre>
<p>Make melt dataframe of these subsetted phyloseq taxa objects</p>
<pre class="r"><code>## Borrelia
Borrelia &lt;- subset_taxa(ps_dec_bact, Genus == &quot;Borrelia&quot;)
Borrelia &lt;- prune_taxa(taxa_sums(Borrelia) &gt; 0, Borrelia)
# melt dataframe into long df
m_Borrelia = psmelt(Borrelia)
#Remove all rows were aundance is zero
m_Borrelia2 = m_Borrelia[m_Borrelia$Abundance != 0, ]
# Summarise
sum &lt;- m_Borrelia2 %&gt;%
  group_by(animalID, SampleCategory, gDNAID) %&gt;%
  summarize(sum_Abund = sum(Abundance))


## Anaplasmataceae
# melt dataframe into long df
m_Anaplasmataceae = psmelt(Anaplasmataceae)
#Remove all rows were aundance is zero
m_Anaplasmataceae2 = m_Anaplasmataceae[m_Anaplasmataceae$Abundance != 0, ]

## Rickettsia
Rickettsia &lt;- subset_taxa(ps_dec_bact, Genus == &quot;Rickettsia&quot;)
Rickettsia &lt;- prune_taxa(taxa_sums(Rickettsia) &gt; 0, Rickettsia)
# melt dataframe into long df
m_Rickettsia = psmelt(Rickettsia)
#Remove all rows were aundance is zero
m_Rickettsia2 = m_Rickettsia[m_Rickettsia$Abundance != 0, ]
# Summarise
sum &lt;- m_Rickettsia2 %&gt;%
  group_by(animalID, SampleCategory, gDNAID) %&gt;%
  summarize(sum_Abund = sum(Abundance))


## Midichloriaceae
Midichloriaceae &lt;- subset_taxa(ps_dec_bact, Family == &quot;Midichloriaceae&quot;)
Midichloriaceae &lt;- prune_taxa(taxa_sums(Midichloriaceae) &gt; 0, Midichloriaceae)
writeXStringSet(Midichloriaceae@refseq, &quot;data/dada2_tois/Midichloriaceae.fa&quot;)
write.csv(m_Midichloriaceae2, &quot;data/dada2_tois/midichloriaceae_melt.csv&quot;)
# melt dataframe into long df
m_Midichloriaceae = psmelt(Midichloriaceae)
#Remove all rows were aundance is zero
m_Midichloriaceae2 = m_Midichloriaceae[m_Midichloriaceae$Abundance != 0, ]

## Rickettsiaceae
# melt dataframe into long df
m_Rickettsiaceae = psmelt(Rickettsiaceae)
#Remove all rows were aundance is zero
m_Rickettsiaceae2 = m_Rickettsiaceae[m_Rickettsiaceae$Abundance != 0, ]

## Coxiellaceae
# melt dataframe into long df
m_Coxiellaceae = psmelt(Coxiellaceae)
#Remove all rows were aundance is zero
m_Coxiellaceae2 = m_Coxiellaceae[m_Coxiellaceae$Abundance != 0, ]

## Mycoplasmataceae
# melt dataframe into long df
m_Mycoplasmataceae = psmelt(Mycoplasmataceae)
#Remove all rows were aundance is zero
m_Mycoplasmataceae2 = m_Mycoplasmataceae[m_Mycoplasmataceae$Abundance != 0, ]

## Francisellaceae
# melt dataframe into long df
m_Francisellaceae = psmelt(Francisellaceae)
#Remove all rows were aundance is zero
m_Francisellaceae2 = m_Francisellaceae[m_Francisellaceae$Abundance != 0, ]

## Diplorickettsiaceae
# melt dataframe into long df
m_Diplorickettsiaceae = psmelt(Diplorickettsiaceae)
#Remove all rows were aundance is zero
m_Diplorickettsiaceae2 = m_Diplorickettsiaceae[m_Diplorickettsiaceae$Abundance != 0, ]


## Bartonella
# melt dataframe into long df
m_Bartonella = psmelt(Bartonella)
#Remove all rows were aundance is zero
m_Bartonella2 = m_Bartonella[m_Bartonella$Abundance != 0, ]

## Merge these datasets
taxa_of_interest = bind_rows(m_Borreliella2, m_Anaplasmataceae2, m_Midichloriaceae2, m_Bartonella2, m_Rickettsiaceae2, m_Diplorickettsiaceae2, m_Francisellaceae2, m_Mycoplasmataceae2, m_Coxiellaceae2)
write_csv(taxa_of_interest, &quot;data/dada2_tois/taxa_of_interest.csv&quot;)

#### EXTRAS

## Rickettsiales
# melt dataframe into long df
m_Rickettsiales = psmelt(Rickettsiales)
#Remove all rows were aundance is zero
m_Rickettsiales2 = m_Rickettsiales[m_Rickettsiales$Abundance != 0, ]

## Mycobacteriaceae
# melt dataframe into long df
m_Mycobacteriaceae = psmelt(Mycobacteriaceae)
#Remove all rows were aundance is zero
m_Mycobacteriaceae2 = m_Mycobacteriaceae[m_Mycobacteriaceae$Abundance != 0, ]

## Eimeria
# melt dataframe into long df
m_Eimeria = psmelt(Eimeria)
#Remove all rows were aundance is zero
m_Eimeria2 = m_Eimeria[m_Eimeria$Abundance != 0, ]</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
