---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

In the spirit of reproducibility this project website is created that outlines analysis done for the following publication:

*The bacterial biome of ticks and their wildlife hosts at the sylvatic urban interface.* 

**Authors**: Siobhon L. Egan, Casey L. Taylor, Peter B. Banks, Amy S. Northover, Liisa A. Ahlstrom, Una M. Ryan, Peter J. Irwin, Charlotte L. Oskam. 

This publication is also a chapater of my PhD Thesis 2021.

Project website available here http://siobhonlegan.com/wildlife-bacteria/ and source code is available at https://github.com/siobhon-egan/wildlife-bacteria/


## Data availbility


Raw Illumina MiSeq data is available at European nucleotide archive the project accession number PRJEB46006 (ERP130188), which includes the following sample accession numbers:

- 16S bacteria blood ERS6632201--ERS6632359
- 16S bacteria tissue ERS6632360--ERS6632531
- 16S bacteria tick ERS6633094--ERS6633353

In addition the following nucleotide sequence data has been deposited to GenBank

- Anaplasmataceae 16S rRNA MW633160--MW633167
- *Borrelia* 16S rRNA MW633074
- *Bartonella* 16S rRNA/Internal Transcribed Spacer region MW633076-MW633082

## Supporting data

Data produced after bioinformatic analysis of raw Illumina MiSeq data such as taxonomy table, ASV count data and metadata is available within the **Rdata/** directory in `.RData` format.

Illumina MiSeq data generated from the metabarcoding of ticks targeting the 12S rRNA locus has been deposited in the European nucleotide archive under the project accession number PRJEB46056 (ERP130244), which includes the following sample accession numbers: ERS6635126--ERS6635348 (BioSample # SAMEA8952359--SAMEA8952582)

Nucleotide data for subset of zOTUs generated are available in for the molecular identification of ticks has been uploaded to GenBank under accession numbers MW665133--MW665150.

---

<!-- Workflowr commands

To build/view webiste locally      
`wflow_build()` and `wflow_view()`

Status of workflow      
`wflow_status()`

Publish website pages       
`wflow_publish(c("analysis/index.Rmd", "analysis/first-analysis.Rmd"), "Add my first analysis")`

Push website to github      
`wflow_git_push()`   -->
